src/app/components/source/UserSecurity.js

```diff
@@ -0,0 +1,522 @@
+        twoFactorAuthentication: two_factor.otp_required,
+        showFactorCommonFields: two_factor.otp_required,
+        showFactorAuthForm: false,
+        backupCodes: [],
+      });
+    };
+    const isValid = this.validateInputs();
+    if (isValid) {
+      const { dbid } = this.props.user;
+      Relay.Store.commitUpdate(
+        new UserTwoFactorAuthenticationMutation({
+          id: dbid,
+          password: this.state.password,
+          otp_required: enabled,
+          qrcode: this.state.qrcode,
+        }),
+        { onSuccess, onFailure },
+      );
+    }
+  }
+
+  handleSecuritySettings(type, e, inputChecked) {
+    const { id } = this.props.user;
+
+    const onFailure = () => {
+    };
+    const onSuccess = () => {
+    };
+    let { sendSuccessfulLogin, sendFailedLogin } = this.state;
+    if (type === 'successfulLogin') {
+      sendSuccessfulLogin = inputChecked;
+    } else {
+      sendFailedLogin = inputChecked;
+    }
+    this.setState({
+      sendSuccessfulLogin,
+      sendFailedLogin,
+    });
+    Relay.Store.commitUpdate(
+      new SetUserSecuritySettingsMutation({
+        id,
+        sendSuccessfulLogin,
+        sendFailedLogin,
+      }),
+      { onSuccess, onFailure },
+    );
+  }
+
+  handleGenerateBackupCodes() {
+    const { dbid } = this.props.user;
+
+    const onFailure = () => {};
+
+    const onSuccess = (response) => {
+      const { generateTwoFactorBackupCodes: { codes } } = response;
+      this.setState({ backupCodes: codes });
+    };
+
+    Relay.Store.commitUpdate(
+      new GenerateTwoFactorBackupCodesMutation({
+        id: dbid,
+      }),
+      { onSuccess, onFailure },
+    );
+  }
+
+  renderMessage(item) {
+    switch (item) {
+    case 'passwordInput':
+      return this.props.intl.formatMessage(messages.passwordInput);
+    case 'passwordError':
+      return this.props.intl.formatMessage(messages.passwordError);
+    case 'verifyInput':
+      return this.props.intl.formatMessage(messages.verifyInput);
+    case 'verifyError':
+      return this.props.intl.formatMessage(messages.verifyError);
+    default:
+      return null;
+    }
+  }
+
+  render() {
+    const { user } = this.props;
+    const { can_enable_otp, qrcode_svg } = user.two_factor;
+    const currentUser = this.getCurrentUser();
+
+    if (!currentUser || !user || currentUser.dbid !== user.dbid) {
+      return null;
+    }
+
+    const style = {
+      margin: `${units(2)} 0`,
+    };
+
+    const cardStyle = {
+      margin: `${units(2)} 0`,
+    };
+
+    const cardTextStyle = {
+      display: 'flex',
+      alignItems: 'center',
+    };
+
+    const cardTextAuthStyle = {
+      display: 'table-row',
+      alignItems: 'center',
+    };
+
+    const subTitleStyle = {
+      margin: '5px 0px',
+      textTransform: 'uppercase',
+    };
+
+    const divBackupStyle = {
+      lineHeight: '25px',
+      fontWeight: 'bold',
+      fontSize: '12px',
+      backgroundColor: '#dfdfe6',
```

Comment:
If you really need a new color, please add it to `src/app/styles/js/shared.js` and use it here... we usually avoid having hard-coded colors in the code

Reply:
use existing one from src/app/styles/js/shared.js